### PR TITLE
add github actions testing

### PR DESCRIPTION
**Pull request by TheTripleV**
Add tests on Github Actions.
This PR tests both backends.
The ntcore backend has failing tests so this will always fail for now.

### CODE REVIEW

**Comment by virtuald** [comment]
I feel if we're going to add github actions, let's remove travis?

**Comment by TheTripleV** [comment]
I agree. I've left out pushing to pypi so far. Are the pypi secrets in this repo?

**Comment by auscompgeek** [comment]
Could we also run the pure Python and pyntcore tests in parallel please?

**Comment by TheTripleV** [comment]
> Could we also run the pure Python and pyntcore tests in parallel please?

Done.
Example run: https://github.com/TheTripleV/pynetworktables/runs/896761369?check_suite_focus=true

File: .github/workflows/dist.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        python -m pip install black
+        black --check --diff  .
+
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
+        python_version: [3.6, 3.7, 3.8]
```
Comment (auscompgeek) [inline .github/workflows/dist.yml:30]: Doesn't pynetworktables give a pure py3 wheel?

Comment (TheTripleV) [reply .github/workflows/dist.yml:30]: Yes. Should it only build on 3.6 then?

Comment (TheTripleV) [reply .github/workflows/dist.yml]: I changed it to only build on 3.6.

**Comment by virtuald** [comment]
Is this good to go? Might be worth using some of the github actions updates I'm pushing out.

**Comment by TheTripleV** [comment]
Yes. The CI is finished.
The only thing left is that a few test cases fail on the ntcore backend: https://github.com/TheTripleV/pynetworktables/runs/934751015?check_suite_focus=true

**Comment by TheTripleV** [comment]
I guess we could use the build actions for building. They can't be used for testing because we need to install pyntcore from tortall here (until linux wheels can be downloaded from pypi).

**Comment by auscompgeek** [comment]
> until linux wheels can be downloaded from pypi

I believe this is actually possible now, we just need to tag our wheels with an appropriate manylinux tag.

**Comment by virtuald** [comment]
It would be good to fix the tests.

> I believe this is actually possible now, we just need to tag our wheels with an appropriate manylinux tag.

Have you done any experimentation with this sort of thing? It would be useful to know if we could do it.

**Comment by virtuald** [comment]
> I guess we could use the build actions for building. They can't be used for testing because we need to install pyntcore from tortall here (until linux wheels can be downloaded from pypi).

I don't understand? The build actions I recently added do pull wheels from tortall?

**Comment by TheTripleV** [comment]
Ah. I'll switch over to using the build actions.

**Comment by virtuald** [comment]
I believe all of the secrets are setup correctly for this repo for the new build actions. The RTD webhook actions aren't necessary for this repo.

**Comment by TheTripleV** [comment]
Some steps now use robotpy build actions.

**Comment by TheTripleV** [comment]
I think the only reason I left this as draft was because of the failing pyntcore test and because the pure stuff is imported all over the place in the tests.

Since actions aren't available here, all pushes to this pr trigger actions here: https://github.com/TheTripleV/pynetworktables/actions

**Pull request merged**